decls/i18n.js

```diff
@@ -121,7 +121,7 @@ declare interface I18n {
   mergeLocaleMessage (locale: Locale, message: LocaleMessageObject): void,
   t (key: Path, ...values: any): TranslateResult,
   i (key: Path, locale: Locale, values: Object): TranslateResult,
-  tc (key: Path, choice?: number, ...values: any): TranslateResult,
+  tc (key: Path, choice?: number, args: any, ...values: any): TranslateResult,
```

Comment:
Yeah, i know, but spread operator...
Can you show example?
I tried, but then we must refactor much code?

Reply:
Thank you for your PR!
And sorry for my late reply  🙇 

As @exoego said, This PR is breaking change.
And we are necessary to understand the concrete use-cases of `getChoiceIndex` a little more. 

if we accept your PR, we have to prepare the document. and we have to major version upgrade is necessary.
As a matter of fact, I think that it is acceptable to receive your PR, if vue-i18n users are needed this PR.

Could you provide the use-case explanation, please? 🙏

Reply:
As i said above `Add args, for make pluralization more flexible, for example suffix rules, countable rules etc.`.
For example:
we need 3 different rules
1 just return string by given index
2 make pluralization with given count
3 do some another action by another rule.

It's usable for example for slavic language's.
`"group": "группа | группы | групп | группу"` key 1. // we need just choose it in getChoiceIndex
`apple: 'no apples | one apple | {count} apples',` // but there we need to provide count
`car: '0 машин | 1 машина | {n} машины | {n} машин',` // there too
`banana: 'нет бананов | 1 банан | {n} банана | {n} бананов'` //there too

From https://kazupon.github.io/vue-i18n/guide/pluralization.html#custom-pluralization :
```js
VueI18n.prototype.getChoiceIndex = function (choice, choicesLength) {
  // this === VueI18n instance, so the locale property also exists here
  if (this.locale !== 'ru') {
    // proceed to the default implementation
  }
  if (choice === 0) {
    return 0;
  }
  const teen = choice > 10 && choice < 20;
  const endsWithOne = choice % 10 === 1;
  if (!teen && endsWithOne) {
    return 1;
  }
  if (!teen && choice % 10 >= 2 && choice % 10 <= 4) {
    return 2;
  }
  return (choicesLength < 4) ? 2 : 3;
}
```
`$tc('path.to.rule', choiceIndex)` we can provide only `choice` and access to vuei18n object which can contain lang and etc.

For example we have 10 different word's in slavic language's in diff cases (nominative), (genitive), (dative), (accusative), (instrumental) but with same meaning.

[In russian lang' there more cases.](https://24smi.org/public/media/resize/800x-/2017/8/10/02_WEucTvk.jpg)
``"group": "группа | группы | групп | группу"` key 1. // we need just choose it in getChoiceIndex`
as you can see `group` = `группа`
`Нет групп` = `No groups`
`Эта группа` = `This group`
`Есть ли здесь группа А?` = `Is there r/n group A?`
`Мы должны посетить эту группу` = `We should visit this group`
`Данной группы нет в списке` = `There no group with this credentials in list`
as you can see in english there one word `group` which can be used with same key in json (translation) file, but in russian we need to create `group` key, but there no another word's for group in english which can describe different cases of this word(instead `groups`, xD).
It's just for usability.
For example.
```js
VueI18n.prototype.getChoiceIndexEng = VueI18n.prototype.getChoiceIndex;//get ptr for original plurazation rules
VueI18n.prototype.getChoiceIndex = function (choice, choicesLength, args) {
  if(args){
    return args.choice; // return choice
  }
  // this === VueI18n instance, so the locale property exists
  if (this.locale !== 'ru'){
	  return i18n.getChoiceIndexEng(choice, choicesLength); //call original function
  }
  
  if (choice === 0) {//do some another stuff
    return 0;
  }

  const teen = choice > 10 && choice < 20;
  const endsWithOne = choice % 10 === 1;

  if (!teen && endsWithOne) {
    return 1;
  }
  if (!teen && choice % 10 >= 2 && choice % 10 <= 4) {
    return 2;
  }
  return (choicesLength < 4) ? 2 : 3;
}
```
So i think it's better to create clear json file with key's like
`"group": "group | groups | groupasd ..."`
instead of
`"groups" : "groups","group":"group"`


Anyway, I don't know how to explain it.
But the 'argument' option should be present everywhere.
I think it's a mistake by https://github.com/kazupon/vue-i18n/pull/482 author who designed the pluralization feature.